### PR TITLE
test: add coverage for protective auction ACs

### PR DESCRIPTION
**Pull request by EVODelavega**
Also included some more unit testing to processor package, changed the `log.Fatal` call in the protocol upgrade flow to a `log.Panic` to be able to recover in a unit test, to check that code-path.

### CODE REVIEW

File: core/execution/future/market.go [hunk 1]
```diff
@@ -1502,6 +1502,10 @@ func (m *Market) EnterLongBlockAuction(ctx context.Context, duration int64) {
 	if !m.canTrade() {
 		return
 	}
+	// markets that are suspended via governance, or are in monitoring auction should be unaffected by long block auctions.
```
Comment (ze97286) [inline core/execution/future/market.go:1505]: you want to drop the second part of the comment

File: core/execution/spot/market.go [hunk 2]
```diff
@@ -492,6 +492,11 @@ func (m *Market) EnterLongBlockAuction(ctx context.Context, duration int64) {
 		return
 	}
 
+	// markets in monitoring or governance auction are unaffected by long block auctions.
```
Comment (ze97286) [inline core/execution/spot/market.go:495]: and the first part here :)

File: core/execution/future/market.go (hunk 1 above)
Comment (EVODelavega) [reply core/execution/future/market.go]: true

File: core/execution/spot/market.go (hunk 2 above)
Comment (EVODelavega) [reply core/execution/spot/market.go]: done

File: core/execution/spot/market.go [hunk 3]
```diff
@@ -492,6 +492,11 @@ func (m *Market) EnterLongBlockAuction(ctx context.Context, duration int64) {
 		return
 	}
 
+	// markets in governance auction are unaffected by long block auctions.
+	if m.mkt.TradingMode == types.MarketTradingModeSuspendedViaGovernance {
+		return
+	}
+
 	m.mkt.State = types.MarketStateSuspended
 	m.mkt.TradingMode = types.MarketTradingModelLongBlockAuction
```
Comment (ze97286) [inline core/execution/spot/market.go:501]: I think this is still wrong you know? or at least out of place. If we're not in an auction then it's correct. But if we are in an auction I'm not sure we should do either of these.

**Review by jeremyletang — APPROVED** [review]

**Pull request merged**